File: packages/compress-stringify/typescript/test/browser/demo-app/compress-stringify-load-sample-inputs.ts
export default async function compressStringifyLoadSampleInputs (model, preRun=false) {
  const sampleInput = new Uint8Array([222, 173, 190, 239])
  model.inputs.set("input", sampleInput)
  if (!preRun) {
    const inputElement = document.getElementById("compressStringify-input-details")
    inputElement.innerHTML = `<pre>${globalThis.escapeHtml(sampleInput.toString())}</pre>`
    inputElement.disabled = false
  }

  const stringify = true
  model.options.set("stringify", stringify)
  if (!preRun) {
    const stringifyElement = document.querySelector('#compressStringifyInputs sl-checkbox[name=stringify]')
    stringifyElement.checked = stringify
  }

  const compressionLevel = 5
  model.options.set("compressionLevel", compressionLevel)
  if (!preRun) {
    const compressionLevelElement = document.querySelector('#compressStringifyInputs sl-input[name=compression-level]')
    compressionLevelElement.value = compressionLevel
  }

  const dataUrlPrefix = 'data:application/iwi+cbor+zstd;base64,'
  model.options.set("dataUrlPrefix", dataUrlPrefix)
  if (!preRun) {
    const dataUrlPrefixElement = document.querySelector('#compressStringifyInputs sl-input[name=data-url-prefix]')
    dataUrlPrefixElement.value = dataUrlPrefix
  }

  return model
}

export const usePreRun = true

File: packages/compress-stringify/python-web-demo/compress_stringify_load_sample_inputs.py
import js

async def load_sample_inputs(model):
    sample_input = bytes([222, 173, 190, 239])
    model.inputs['input'] = sample_input
    input_element = js.document.getElementById('compress_stringify-input-details')
    input_element.innerHTML = f"<pre>{str(sample_input)}</pre>"
    input_element.disabled = False

    stringify = True
    model.options['stringify'] = stringify
    stringify_element = js.document.querySelector('#compress_stringify-inputs sl-checkbox[name=stringify]')
    stringify_element.checked = stringify

    compression_level = 5
    model.options['compression_level'] = compression_level
    compression_level_element = js.document.querySelector('#compress_stringify-inputs sl-input[name=compression-level]')
    compression_level_element.value = compression_level

    data_url_prefix = 'data:application/iwi+cbor+zstd;base64,'
    model.options['data_url_prefix'] = data_url_prefix
    data_url_prefix_element = js.document.querySelector('#compress_stringify-inputs sl-input[name=data-url-prefix]')
    data_url_prefix_element.value = data_url_prefix

    return model